pkg/frontend/querymiddleware/querysharding.go

```diff
@@ -160,17 +164,25 @@ func (s *querySharding) Do(ctx context.Context, r MetricsQueryRequest) (Response
 	if err != nil {
 		return nil, mapEngineError(err)
 	}
+	// Note that the positions based on the original query may be wrong as the rewritten
+	// query which is actually used is different, but the user does not see the rewritten
+	// query, so we pass in an empty string as the query so the positions will be hidden.
```

Comment:
Seems like a bug we should track in an issue. Ie: Reconstruct annotation positions after sharding.

Reply:
I've created https://github.com/grafana/mimir/issues/9141 to track this